WordPress/wordpress-develop
# Backport: Store legacy sidebars when switching to a block theme

Trac ticket: https://core.trac.wordpress.org/ticket/57531
Related Gutenberg PR: https://github.com/WordPress/gutenberg/pull/45509

## Testing instructions without package updates

1. Switch to the Twenty Twenty theme.
2. Confirm sidebars have at least one widget by going to Appearance > Widgets
3. Switch to any block theme. I was using Twenty Twenty-Three in my tests.
4. Make a get request to the sidebars endpoint. It requires auth, but the Application Password could be used for basic auth.
5. Confirm that the request loads the same sidebars registered for the Twenty Twenty theme.

**Example request made by the editor.**

```http
GET /wp-json/wp/v2/sidebars?context=edit&_fields=id,name,description,status,widgets
```

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**
